core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -321,6 +320,10 @@ public T get(Settings settings) {
     public String getRaw(Settings settings) {
         // They're using the setting, so we need to tell them to stop
         if (this.isDeprecated() && this.exists(settings)) {
+            // it does not matter if this is set twice
+            if (deprecationLogger == null) {
+                deprecationLogger = new DeprecationLogger(Loggers.getLogger(Setting.class));
```

Comment:
Given the need and the rare use of particular type of deprecation code, perhaps we should just construct a new `DeprecationLogger` on the fly and not have it be `static` at all? Just create it, use it, then throw it away?